pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -608,6 +609,11 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 		gracePeriod = *pod.DeletionGracePeriodSeconds
 	case pod.Spec.TerminationGracePeriodSeconds != nil:
 		gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
+		if annotationGracePeriod, found := pod.ObjectMeta.Annotations["unsupported.do-not-use.openshift.io/override-liveness-grace-period-seconds"]; found {
+			if val, err := strconv.Atoi(annotationGracePeriod); err == nil {
+				gracePeriod = int64(val)
```

Comment:
How do we want to validate this?

Reply:
we can add an e2e over in openshift/origin

Reply:
We will likely want to use ParseUint so the gracePeriod isn't negative. I don't think there is a range we need to keep in mind.

Reply:
[0..n] would be sufficient.

Reply:
I added a > 0 check out of habit, but I guess it's possible we'd want immediate termination rather than 1s? Open to changing :)